src/resources/schema/document-code.yml

```diff
@@ -45,6 +48,7 @@
     long: |
       The style to use when displaying code annotations. Set this value
       to false to hide code annotations.
+      Limitation: does not work properly with `code-link` set to `true`.
```

Comment:
Same here are for `code-line-numbers`. 

I believe code annotations works ok - it is just that there will be no code linking applied by **downlit**

Try this 

````markdown
---
format: 
  html:
    code-link: true
    code-annotations: hover
---

```{r}
#| message: false
library(dplyr) # <1>
first(mtcars) # <2>
last(1:10)
```

1. Test
2. Test
````

and this 

````markdown
---
format: 
  html:
    code-link: true
---

```{r}
#| message: false
library(dplyr)
first(mtcars)
last(1:10)
```
````

I don't think noting that `code-annotations` does not work properly is the same. This is `code-link` that won't work.

Reply:
See <https://github.com/quarto-dev/quarto-cli/discussions/6187>.

I would not say discarding a parameter and outputting a warning is the definition of `code-annotations` and `code-link` working properly.

Reply:
How about changing to:
> When `code-annotations` is set to `true`, `code-link` is automatically disabled due to downlit current limitations (*for HTML-based formats using knitr as the engine*).

Reply:
I saw #6187  - there is no example in it, and nothing that shows that Code Annotations feature is not working as it should. 

> I would not say discarding a parameter and outputting a warning is the definition of code-annotations and code-link working properly.

And I did not say (or at least meant) that both are working correctly. I am saying that `code-annotations` is working ok but if you add `code-link` then we know code linking won't work. So we emit a warning so that the user know when rendering (and not discover later that no code linking is in the doc). 

To me the limitation is to `code-link` and the information should be in this reference description. It will be only shown on HTML Reference, and really is the limitation. 

> It's a failsafe. User should not have to be aware of that after the facts.

I am not sure what we could do more when two features don't work together than 

* do not error and proceed with deactivating the conflicting feature
* Warn user so that it is known and not silently tweaked
* Document `code-link` limitation. 

Overall, we should indeed document limitations - that is the purpose of main epic https://github.com/quarto-dev/quarto-cli/issues/6161

Just the place of it seems to me important, to not give the impression that a feature does not work. (and by working I mean the individual feature doing what it is supposed to do - not no warning whatsoever)

Reply:
I agree, let's only document these limitation under `code-link`.

Reply:
reverted in ad6c8f7c42b850d666403423c6d6a275a654c3bb